src/libcore/iter/iterator.rs

```diff
@@ -2463,6 +2463,52 @@ pub trait Iterator {
             }
         }
     }
+
+    /// Extends passed collection with the contents of an iterator.
+    ///
+    /// This method consumes the iterator and adds all its items to the
+    /// passed collection. The collection is then returned, so the call chain
+    /// can be continued.
+    ///
+    /// This method is a counter-part of `Extend::extend`, but instead of
+    /// being called on collection, it's called on iterator.
+    ///
+    /// ```
+    /// let result = (3..5).extend(vec![1, 2]);
+    /// assert_eq!(vec![1, 2, 3, 4], result);
+    /// ```
+    #[unstable(feature = "iterator_extend", issue = "0")]
+    fn extend<E>(self, mut extended: E) -> E
+        where
+            E: Extend<Self::Item>,
+            Self: Sized {
```

Comment:
There's quite a lot of indentation here - run rustfmt?

Reply:
I ran rustfmt `0.3.8-nightly (346238f 2018-02-04)` on libcore, but the results were rather catastrophic. It threw few dozens of errors, even more warnings and modified 950 files. I think I'll just manually format it to fit the convention of other methods.

Reply:
@CodeSandwich You could copy only this function into the [playground](https://play.rust-lang.org) and format it, so that you don't need to care about the rest of libcore.

![mar-01-2018 05-25-11](https://user-images.githubusercontent.com/103023/36813955-25b4b0e4-1d11-11e8-9270-9c80b308221d.gif)